File: 00 C++ Intro/Jumping/a62_q1a_jumping.py
N = int(input())
A = [int(e) for e in input().split()]
SUM = []
for i in range(N):
    if(i == 0):
        SUM.append(A[i])
    elif(i == 1):
        SUM.append(SUM[i-1] + A[i])
    elif(i == 2):
        SUM.append(max(SUM[i-2], SUM[i-1]) + A[i]) 
    else:
        SUM.append(max(SUM[i-3], SUM[i-2], SUM[i-1]) + A[i])
print(SUM[-1])

File: 00 C++ Intro/Jumping/a62_q1a_jumping.cpp
#include<iostream>
#include<vector>
using namespace std;
int N;
vector<int> A, SUM;
int main()
{
    cin >> N;
    for(int i = 0; i < N; i++)
    {
        int temp;
        cin >> temp;
        A.push_back(temp);
        if(i == 0) { SUM.push_back(A[i]); }
        else if(i == 1) { SUM.push_back(SUM[i-1] + A[i]); }
        else if(i == 2)
        {
            temp = max(SUM[i-2], SUM[i-1]) + A[i];
            SUM.push_back(temp);
        }
        else
        {
            temp = max( SUM[i-3], max(SUM[i-2], SUM[i-1]) ) + A[i];
            SUM.push_back(temp);
        }
    }
    cout << SUM[N-1];
}